manywheel/build_rocm.sh

```diff
@@ -223,56 +213,86 @@ HIPBLASLT_LIB_FILES=($ARCH_SPECIFIC_FILES $OTHER_FILES)
 # ROCm library files
 # Overwrite ROCM_SO_FILES to contain only libmagma.so if BUILD_LIGHTWEIGHT is enabled
 if [[ "$BUILD_LIGHTWEIGHT" == "1" ]]; then
-    ROCM_SO_FILES=(
+    LIGHTWEIGHT_ROCM_SO_FILES=(
         "libmagma.so"
     )
 fi
-ROCM_SO_PATHS=()
-for lib in "${ROCM_SO_FILES[@]}"
-do
-    file_path=($(find $ROCM_HOME/lib/ -name "$lib")) # First search in lib
-    if [[ -z $file_path ]]; then
-        if [ -d "$ROCM_HOME/lib64/" ]; then
-            file_path=($(find $ROCM_HOME/lib64/ -name "$lib")) # Then search in lib64
+
+do_lightweight_build() {
+    echo "=== Building LIGHTWEIGHT variant ==="
+
+    # First, gather actual file paths for the minimal set
+    ROCM_SO_PATHS_LIGHTWEIGHT=()
+    for lib in "${LIGHTWEIGHT_ROCM_SO_FILES[@]}"; do
+        file_path=($(find $ROCM_HOME/lib/ -name "$lib"))
+        if [[ -z $file_path && -d "$ROCM_HOME/lib64" ]]; then
+             file_path=($(find $ROCM_HOME/lib64/ -name "$lib")) # Then search in lib64
         fi
+        if [[ -z $file_path ]]; then
+            file_path=($(find $ROCM_HOME/ -name "$lib")) # Then search in ROCM_HOME
+        fi
+        if [[ -z $file_path ]]; then
+            echo "Error: Lightweight library $lib is not found." >&2
+            exit 1
+        fi
+	ROCM_SO_PATHS_LIGHTWEIGHT[${#LIGHTWEIGHT_ROCM_SO_FILES[@]}]="$file_path" # Append lib to array
```

Comment:
Why does this not use `ROCM_SO_PATHS_LIGHTWEIGHT[${#ROCM_SO_PATHS_LIGHTWEIGHT[@]}]`? The point is to append `$file_path` to the end of the `ROCM_SO_PATHS_LIGHTWEIGHT` array, right?